src/__smoke-test__/smoke-test.spec.ts

```diff
@@ -0,0 +1,446 @@
+import { test, expect } from '@playwright/test';
+import {
+    HTTP_EVENT_TYPE,
+    JS_ERROR_EVENT_TYPE,
+    LCP_EVENT_TYPE,
+    FID_EVENT_TYPE,
+    PERFORMANCE_NAVIGATION_EVENT_TYPE,
+    PERFORMANCE_RESOURCE_EVENT_TYPE,
+    PAGE_VIEW_EVENT_TYPE,
+    SESSION_START_EVENT_TYPE
+} from '../plugins/utils/constant';
+
+// Secrete constant values retrieved from GitHub
+const endpoint = '${process.env.ENDPOINT}';
+const monitorId = '${process.env.MONITOR_ID}';
+const url = '${process.env.SMOKE_URL}';
+
+function parseBatches(requestData) {
+    const output = [];
+    for (let i = 0; i < requestData.length; i++) {
+        const parsed = JSON.parse(requestData[i]).RumEvents;
+        for (let j = 0; j < parsed.length; j++) {
+            output.push(parsed[j]);
+        }
+    }
+    return output;
+}
+
+function addEventsToList(parsedPayload, eventList) {
+    for (let i = 0; i < parsedPayload.length; i++) {
+        eventList.push(parsedPayload[i].id);
+    }
+}
+
+function getEventsByType(requestBody, eventType) {
+    return requestBody.filter((e) => e.type === eventType);
+}
+
+function filterSuccessfulResponse(response, requestBody) {
+    // filter out irrelevant, unsuccessful network request/response
+    const request = response.request();
+    if (
+        request &&
+        response.status() === 200 &&
+        request.method() === 'POST' &&
+        response.url() === endpoint + monitorId + '/' &&
+        request.postData()
+    ) {
+        requestBody.push(request.postData());
+    }
+}
+
+// Timestamp to filter events fetched via getAppMonitorData API
+const time = Date.now();
+
+test.describe('Smoke test fixtures', () => {
+    let page;
+    const eventIds = [];
+
+    // Use single browser context to prevent multiple sessions per test
+    test.beforeAll(async ({ browser }) => {
+        const context = await browser.newContext();
+        page = await context.newPage();
+        await page.goto(url);
+    });
+
+    test.afterAll(async ({ browser }) => {
+        browser.close;
+    });
+
+    test('test initial events are emitted', async () => {
+        const SESSION_START_COUNT = 1;
+        const PAGEVIEW_COUNT = 1;
+        const NAVIGATION_COUNT = 1;
+        const RESOURCE_EVENT_COUNT = 1;
+        const expectedPage = '/smoke.html';
+        const requestBody = [];
+
+        // Collect successful requests
+        page.on('response', (response) =>
+            filterSuccessfulResponse(response, requestBody)
+        );
+
+        // Wait until fetch is complete
+        await page.waitForTimeout(10000);
```

Comment:
Can we use waitFor (event to happen) instead of sleep? Sleep create nondeterminism (flakiness) and increases the length of time the test runs.

Reply:
Didn't know waitForTimeout has those side effects. Just looked into the official documentation, I think we could use waitForRequest or waitForResponse to have the same effect. Will reflect in next revision.